pkg/storage/pv.go

```diff
@@ -60,6 +74,78 @@ func (builder *PVBuilder) Exists() bool {
 	return err == nil || !k8serrors.IsNotFound(err)
 }
 
+// Delete removes a PersistentVolume from the apiClient if it exists.
+func (builder *PVBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the PersistentVolume %s", builder.Definition.Name)
+
+	if !builder.Exists() {
+		glog.V(100).Infof("PersistentVolume %s cannot be deleted because it does not exist", builder.Definition.Name)
+
+		builder.Object = nil
+
+		return nil
+	}
+
+	err := builder.apiClient.PersistentVolumes().Delete(context.TODO(), builder.Definition.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	builder.Object = nil
+
+	return nil
+}
+
+// DeleteAndWait deletes the PersistentVolume and waits up to timeout until it has been removed.
+func (builder *PVBuilder) DeleteAndWait(timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof(
+		"Deleting PersistentVolume %s and waiting up to %s until it is removed", timeout, builder.Definition.Name)
+
+	err := builder.Delete()
+	if err != nil {
+		return err
+	}
+
+	return builder.WaitUntilDeleted(timeout)
+}
+
+// WaitUntilDeleted waits for the duration of timeout or until the PersistentVolume has been deleted.
+func (builder *PVBuilder) WaitUntilDeleted(timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Waiting up to %s until PersistentVolume %s is deleted", timeout, builder.Definition.Name)
+
+	return wait.PollUntilContextTimeout(
+		context.TODO(), time.Second, timeout, true, func(ctx context.Context) (bool, error) {
+			_, err := builder.apiClient.PersistentVolumes().Get(context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+			if err == nil {
+				glog.V(100).Infof("PersistentVolume %s still present", builder.Definition.Name)
+
+				return false, nil
+			}
+
+			if k8serrors.IsNotFound(err) {
+				glog.V(100).Infof("PersistentVolume %s is gone", builder.Definition.Name)
+
+				return true, nil
+			}
+
+			glog.V(100).Infof("failed to get PersistentVolume %s", builder.Definition.Name)
+
+			return false, err
```

Comment:
Could this be simplified into
```
return builder.Exists(), nil
```

Reply:
You'd lose some logging and the ability to differentiate errors that don't come from the PV not being found, I'm not sure how much we care about the other errors being returned though